TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/TowerController.java

```diff
@@ -45,5 +67,35 @@ public void handleScrew() {
             }
         }
     }
+
+    public void handleUBar() {
+        //set U bar to position depending on the uBarNum variable
+        if (uBarNum == 1) {
+            drive(COUNTS_PER_INCH * 20, 1);
+        }
+        if (uBarNum == 2) {
+            drive(COUNTS_PER_INCH * 40, 1);
+        }
+        if (uBarNum == 3) {
+            drive(COUNTS_PER_INCH * 60, 1);
+        }
+        if (uBarNum == 4) {
+            drive(COUNTS_PER_INCH * 80, 1);
+        }
+    }
+
+    private void drive(double uBarTarget, double speed) {
+        uBarLevel += uBarTarget;
+        uBar.setTargetPosition(uBarLevel);
+    }
+
+    public void handleIntake (HardwareMap hardwareMap) {
+        if(intakePos){
+            intake.setPosition(1);
+        }
+        else{
+            intake.setPosition(0);
+        }
+    }
 }
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        //hi. you found me. -SECRET COMMENT
```

Comment:
I think you should make a handleTower() function that handles all the components of the tower: handleIntake, handleBar, handleScrew.

Reply:
No, that's for our main code for TeleOp